### PR TITLE
Allow to turn off the sicky mode by setting v-sticky="null"

### PR DESCRIPTION
**Pull request by 0m4r**
Allow to turn off the sticky mode by setting v-sticky="null" and to
restore the sticky behaviour by setting a new configuration object.
Example page updated to show the feature

### CODE REVIEW

**Comment by 0m4r** [comment]
@rguanghui any chance you have a look at this pull request?

File: src/index.js [hunk 1]
```diff
@@ -20,8 +25,14 @@ export default {
...
 
-    let childStyle = el.firstElementChild.style
-    childStyle.cssText = `left: 0; right: 0; top: ${stickyTop}px; z-index: ${zIndex}; ${childStyle.cssText}`
+    if (el.firstElementChild) {
+      childStyle = el.firstElementChild.style
+      childStyle.cssText = `left: 0; right: 0; top: ${stickyTop}px; z-index: ${zIndex}; ${childStyle.cssText}`
+    } else {
+      if (window.console && window.console.warn) {
+        window.console.warn('el.firstElementChild is undefined or null', el)
+      }
```
Comment (iqingting) [inline src/index.js]: it's unnecessary to judgment window.console

File: src/index.js [hunk 2]
```diff
@@ -1,14 +1,19 @@
 let listenAction
 let stickyTop
 let zIndex
+let className
+let childStyle
```
Comment (iqingting) [inline src/index.js]: define under the line to use is better

File: src/index.js [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
   bind(el, binding) {
+    if (!binding.value) {
+      return
+    }
     const elStyle = el.style
     const params = binding.value || {}
     stickyTop = params.stickyTop || 0
     zIndex = params.zIndex || 1000
-
+    className = params.className || 'vue-sticky'
```
Comment (iqingting) [inline src/index.js]: it's unnecessary, you can use like

`<div v-sticky class="xx"></div>`

File: src/index.js [hunk 4]
```diff
@@ -32,21 +43,26 @@ export default {
...
+      if (childStyle) {
+        childStyle.position = ''
+      }
       active = false
     }
 
     const check = () => {
       const offsetTop = el.getBoundingClientRect().top
-      if (offsetTop <= stickyTop) {
+      if (stickyTop !== null && offsetTop <= stickyTop) {
```
Comment (iqingting) [inline src/index.js]: `stickyTop = params.stickyTop || 0`

so this cannot be `null`

Comment (0m4r) [reply src/index.js]: well, with the changes I have made you could set the ```v-sticky=null``` to turn it off so, ```stickyTop``` could actually be ```null```

File: src/index.js (hunk 1 above)
Comment (0m4r) [reply src/index.js]: not sure I get this, you say it is unnecessary to check for the availability of ```window.console``` or to output the log itself?

**Comment by iqingting** [comment]
`v-sticky="{ disabled: true }"`

I think it's a better config

**Comment by 0m4r** [comment]
@rguanghui I have updated the pull request. it should also fix https://github.com/rguanghui/vue-sticky/issues/19

Would you have another look?

**Comment by iqingting** [comment]
3.3.4 published

**Pull request merged**